src/main/java/uk/gov/pay/directdebit/mandate/services/OneOffMandateService.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.pay.directdebit.mandate.services;
+
+import javax.inject.Inject;
+import uk.gov.pay.directdebit.gatewayaccounts.model.GatewayAccount;
+import uk.gov.pay.directdebit.mandate.api.ConfirmMandateRequest;
+import uk.gov.pay.directdebit.mandate.model.Mandate;
+import uk.gov.pay.directdebit.mandate.model.OneOffConfirmationDetails;
+import uk.gov.pay.directdebit.payers.model.AccountNumber;
+import uk.gov.pay.directdebit.payers.model.BankAccountDetails;
+import uk.gov.pay.directdebit.payers.model.SortCode;
+import uk.gov.pay.directdebit.payments.api.CreatePaymentRequest;
+import uk.gov.pay.directdebit.payments.model.PaymentProviderFactory;
+import uk.gov.pay.directdebit.payments.model.Transaction;
+import uk.gov.pay.directdebit.payments.services.TransactionService;
+
+public class OneOffMandateService implements MandateCommandService {
+    private final PaymentProviderFactory paymentProviderFactory;
+    private final TransactionService transactionService;
+    private final MandateStateUpdateService mandateStateUpdateService;
+    private final MandateService mandateService;
+    @Inject
+    public OneOffMandateService(
+            PaymentProviderFactory paymentProviderFactory,
+            TransactionService transactionService,
+            MandateStateUpdateService mandateStateUpdateService,
+            MandateService mandateService){
+        this.paymentProviderFactory = paymentProviderFactory;
+        this.transactionService = transactionService;
+        this.mandateStateUpdateService = mandateStateUpdateService;
+        this.mandateService = mandateService;
+    }
+    
+    public Transaction create(GatewayAccount gatewayAccount,
```

Comment:
I find it strange that we create a mandate but return a transaction

Reply:
true, could return both but mandate is not needed upstream, so a bit of a weird one

Reply:
well, we currently only need the transaction, but still seems more natural for it to return the thing it created, then if the calling context needs the transaction, it can get it from the mandate. But fine with it as is